### PR TITLE
[New Unit] Lottery Tank

### PR DESCRIPTION
**Pull request by JasonLannel**
一个基于初始模型的简单“抽奖”坦克。有不同的子弹buff，仿照某些游戏的抽卡机制设计了抽到不同子弹buff的机制。
第一种是子弹伤害倍率变大（10倍），20%几率抽到，10次未抽中开始增加概率，第20次必中。
第二种是子弹变为散射（5发），2%几率抽到，20次未抽中开始增加概率，第50次必中。
开火间隔为 1s。

### CODE REVIEW

File: src/battle_game/core/units/lottery_tank.cpp [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  battle_game::DrawModel(tank_turret_model_index);
+}
+
+void LotteryTank::Update() {
+  TankMove(3.0f, glm::radians(180.0f));
+  TurretRotate();
+  Fire();
+}
+
+void LotteryTank::TankMove(float move_speed, float rotate_angular_speed) {
```
Comment (LazyJazz) [inline src/battle_game/core/units/lottery_tank.cpp:88]: Please inherit from unit::Tank if you are reusing these basic functions.

**Review by LazyJazz — CHANGES_REQUESTED** [review]

File: src/battle_game/core/units/lottery_tank.cpp (hunk 1 above)
Comment (LazyJazz) [reply src/battle_game/core/units/lottery_tank.cpp:88]: You still can override the functions you are going to change even if you are inheriting from unit::Tank.

Comment (JasonLannel) [reply src/battle_game/core/units/lottery_tank.cpp]: Thanks. I've just fixed my codes.

File: src/battle_game/core/units/lottery_tank.cpp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                     BigBulletCeil);
+}
+
+bool LotteryTank::IsScatter() {
+  return CardLottery(GetOnceTimes, InitProb_Scatter, ScatterBulletFloor,
+                     ScatterBulletCeil);
+}
+
+LotteryTank::BulletType LotteryTank::GetBulletType() {
+  uint32_t Type = IsBig() | (IsScatter() << 1);
```
Comment (LazyJazz) [inline src/battle_game/core/units/lottery_tank.cpp:65]: variable name should be in lower_case

File: src/battle_game/core/units/lottery_tank.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  // Lottery Probability Settings
+  uint32_t GetNormalTimes{0};
+  uint32_t GetOnceTimes{0};
+  const uint32_t BigBulletFloor{10};
+  const uint32_t BigBulletCeil{20};
+  const uint32_t ScatterBulletFloor{20};
+  const uint32_t ScatterBulletCeil{50};
+  const float InitProb_Big{0.2f};
+  const float InitProb_Scatter{0.02f};
```
Comment (LazyJazz) [inline src/battle_game/core/units/lottery_tank.h:46]: use lower_case_ with an underscore in the end for private variables

**Review by LazyJazz — CHANGES_REQUESTED** [review]

**Comment by JasonLannel** [comment]
Thanks. I've just fixed the format and support passive skills & bullets with infos by using the Skill structure.

**Review by LazyJazz — DISMISSED** [review]
LGTM! Call me in wechat if you want to merge your contribution into the main branch.

**Pull request closed without merging**